src/integrations/prefect-slack/prefect_slack/credentials.py

```diff
@@ -127,9 +119,13 @@ async def notify_async(self, body: str, subject: Optional[str] = None):
         """
         Sends a message to the Slack channel.
         """
-        await _notify_async(self, body, subject)
+        client = self.get_client()
+
+        response = await client.send(text=body)
+
+        self._raise_on_failure(response)
 
-    @async_dispatch(_notify_async)  # type: ignore
+    @async_dispatch(notify_async)
```

Comment:
i think i'm having a senior moment - why does this work?

Reply:
This also threw me for a loop. From what I've gathered from Python's [descriptor protocol docs](https://docs.python.org/3/howto/descriptor.html#functions-and-methods), `notify_async` is available as an unbound function in this context. The function isn't bound until the class is used, but since we're passing `self` along with the other args and kwargs through `async_dispatch`, `notify_async` is able to function as a bound function as expected. TIL!

Reply:
there's a problem with the binding wrt the properties to force one way or the other that I am trying to articulate with some tests now